sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -1588,19 +1594,63 @@ public void setProfilesSampleRate(final @Nullable Double profilesSampleRate) {
   /**
    * Returns a list of origins to which `sentry-trace` header should be sent in HTTP integrations.
    *
+   * @deprecated use {{@link SentryOptions#getTracePropagationTargets()} }
    * @return the list of origins
    */
+  @Deprecated
+  @SuppressWarnings("InlineMeSuggester")
   public @NotNull List<String> getTracingOrigins() {
-    return tracingOrigins;
+    return getTracePropagationTargets();
   }
 
   /**
    * Adds an origin to which `sentry-trace` header should be sent in HTTP integrations.
    *
+   * @deprecated use {{@link SentryOptions#addTracePropagationTarget(String)} }
    * @param tracingOrigin - the tracing origin
    */
+  @Deprecated
+  @SuppressWarnings("InlineMeSuggester")
   public void addTracingOrigin(final @NotNull String tracingOrigin) {
-    this.tracingOrigins.add(tracingOrigin);
+    addTracePropagationTarget(tracingOrigin);
+  }
+
+  @Deprecated
+  @SuppressWarnings("InlineMeSuggester")
+  @ApiStatus.Internal
+  public void setTracingOrigins(List<String> tracingOrigins) {
```

Comment:
Do we even need this one, if it wasn't here before?

Reply:
Without it the test for supporting the `tracing-origins` parsing for Spring Boot failed because it couldn't find the setter. Didn't investigate why that is. Seemed like the easiest way out to just add it and instantly deprecate. I could probably move it to `SentryProperties` or investigate if you prefer.

Reply:
I'd not add a new public API as `@Deprecated`, I'd rather fix the issue if there's one using the new origins.

Reply:
Not sure how this broke. Can investigate for a bit but would prefer not to delay things because of it.

Reply:
Ah just had a talk with @lbloder. Due to changing the return value of the getter to unmodifiable list Spring is no longer able to use the getter and modify the list directly.

To avoid the unmodifiable list we'd have to use a list where the default entry is in it. The probem then becomes how and when we remove the default entry. For Spring we don't have control over the behaviour as it is done automatically and there's no Sentry code like we have for Manifest and ExternalOptions where we can check whether any value was set and can then decide whether to use the default.